src/main/java/org/graylog/aws/inputs/transports/KinesisTransport.java

```diff
@@ -44,38 +54,98 @@
     private static final String CK_SECRET_KEY = "aws_secret_key";
     private static final String CK_ASSUME_ROLE_ARN = "aws_assume_role_arn";
     private static final String CK_KINESIS_STREAM_NAME = "kinesis_stream_name";
+    private static final String CK_KINESIS_RECORD_BATCH_SIZE = "kinesis_record_batch_size";
+    private static final String CK_KINESIS_MAX_THROTTLED_WAIT_MS = "kinesis_max_throttled_wait";
+
+    public static final int DEFAULT_BATCH_SIZE = 10000;
+    public static final int DEFAULT_THROTTLED_WAIT_MS = 60000;
+    public static final int KINESIS_CONSUMER_STOP_WAIT_MS = 15000;
 
     private final Configuration configuration;
     private final org.graylog2.Configuration graylogConfiguration;
     private final NodeId nodeId;
     private final LocalMetricRegistry localRegistry;
     private final ClusterConfigService clusterConfigService;
+    private final ObjectMapper objectMapper;
 
     private KinesisConsumer reader;
+    ExecutorService executor = null;
+    public Future<?> kinesisTaskFuture = null;
+
+    /**
+     * Indicates if the Kinesis consumer has been stopped due to throttling. Allows the consumer to be restarted
+     * once throttling is cleared.
+     */
+    public AtomicBoolean stoppedDueToThrottling = new AtomicBoolean(false);
+    public KinesisTransportState consumerState = STOPPED;
```

Comment:
Shouldn't `consumerState` also be an `AtomicReference`? It will be set from the `KinesisConsumer` and that runs in a different thread.

Reply:
Fixed. Initialization now done in constructor.